lib/v2/xiaohongshu/util.js

```diff
@@ -65,7 +66,104 @@ const getBoard = (url, cache) =>
         false
     );
 
+const getNotes = (url, cache) =>
+    cache.tryGet(
+        url + '/notes', // To avoid mixing with the cache for `user.js`
+        async () => {
+            let user = '';
+            let notes = [];
+
+            const browser = await require('@/utils/puppeteer')({ stealth: true });
+            try {
+                const page = await browser.newPage();
+                logger.debug(`Requesting ${url}`);
+                await page.goto(url);
+
+                const [otherInfo, userPosted] = await Promise.all(
+                    ['/api/sns/web/v1/user/otherinfo', '/api/sns/web/v1/user_posted'].map((p) =>
+                        page
+                            .waitForResponse((res) => {
+                                const req = res.request();
+                                return req.url().includes(p) && req.method() === 'GET';
+                            })
+                            .then((r) => r.json())
+                    )
+                );
+
+                // Get full text for each note
+                const notesPromise = userPosted.data.notes.map(async (n) => {
```

Comment:
Please each note separately with a longer TTL instead of `config.cache.routeExpire`